dts/arm/nxp/nxp_lpc55S0x_common.dtsi

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) 2022 metraTec
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arm/armv8-m.dtsi>
+#include <dt-bindings/clock/mcux_lpc_syscon_clock.h>
+#include <dt-bindings/gpio/gpio.h>
+#include <mem.h>
+
+/ {
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu@0 {
+			compatible = "arm,cortex-m33f";
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			mpu: mpu@e000ed90 {
+				compatible = "arm,armv8m-mpu";
+				reg = <0xe000ed90 0x40>;
+				arm,num-mpu-regions = <8>;
+			};
+		};
+	};
+};
+
+&sram {
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	sramx: memory@4000000 {
+		compatible = "mmio-sram";
+		reg = <0x4000000 DT_SIZE_K(16)>;
+	};
+
+	sram0: memory@20000000 {
+		compatible = "mmio-sram";
+		reg = <0x20000000 DT_SIZE_K(32)>;
+	};
+
+	sram1: memory@20008000 {
+		compatible = "mmio-sram";
```

Comment:
changing the `compatible = "mmio-sram"` nodes to `compatible = "mmio-sram", "zephyr,memory-region"`  should fix your CI build failure. See #42008 for examples of the required change